docs/ckeditor/plugin.md

```diff
@@ -0,0 +1,504 @@
+# Adding a custom plugin to CKEditor
+---
+
+## Contents
+
+1. [Writing the Editing Plugin](#writing-the-editing-plugin)
+    1. [Define a new plugin](#define-a-new-plugin)
+    1. [Define the schema](#define-the-schema)
+    1. [Define converters](#define-converters)
+    1. [Putting the editing plugin together](#putting-the-editing-plugin-together)
+1. [Writing the UI plugin](#writing-the-ui-plugin)
+    1. [Element creator helper function](#element-creator-helper-function)
+    1. [Element creator `Command`](#element-creator-command)
+    1. [Putting the UI plugin together](#putting-the-ui-plugin-together)
+1. [Pulling it all together](#pulling-it-all-together)
+    1. [Adding a glue plugin](#Adding-a-glue-plugin)
+    1. [Configuring CKEditor](#Configuring-CKEditor)
+        1. [Adding the plugin](#Adding-the-plugin)
+        1. [Configure the toolbar](#configure-the-toolbar)
+
+---
+
+This document describes how to add a custom plugin to CKEditor, using the example of embedding a Youtube video as a simple example.
+
+A complete example show all the code from this example can be found here: <https://github.com/mitodl/ocw-studio/commit/25f57c11540fd001a985b38fdd50a321fba67a4e>
```

Comment:
Nit: 
```suggestion
A complete example with the full implementation described in this doc can be found here: <https://github.com/mitodl/ocw-studio/commit/25f57c11540fd001a985b38fdd50a321fba67a4e>
```

Reply:
not gonna lie I didn't re-read that after typing :joy: